token/program-2022/src/pod.rs

```diff
@@ -95,16 +110,69 @@ impl From<OptionalNonZeroElGamalPubkey> for Option<ElGamalPubkey> {
         }
     }
 }
+#[cfg(feature = "serde-traits")]
+impl Serialize for OptionalNonZeroElGamalPubkey {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        if self.0 == ElGamalPubkey::default() {
+            s.serialize_none()
+        } else {
+            s.serialize_some(&self.0.to_string())
+        }
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+struct OptionalNonZeroElGamalPubkeyVisitor;
+#[cfg(feature = "serde-traits")]
+impl<'de> Visitor<'de> for OptionalNonZeroElGamalPubkeyVisitor {
+    type Value = OptionalNonZeroElGamalPubkey;
```

Comment:
Curious why the need for a new struct instead of implementing `Visitor` directly on `OptionalNonZeroElGamalPubkey` and making `Value` of type `String` ?

Reply:
@buffalojoec 

I implemented `Visitor` on a new struct because as far as I can tell it's the common serde pattern but I can move it on `OptionalNonZeroElGamalPubkey` no problem.

As far as making `Value` of type `String`, could you help me by elaborating a bit? The deserializer converts serialized data into `OptionalNonZeroElGamalPubkey` so this is why I set `type Value = OptionalNonZeroElGamalPubkey;`.

Reply:
Actually I think what you have is correct, my bad! (I had to go read some more on `Visitor`) 
Resolving